_posts/2015-03-30-geomesa-feature-level-visibility.md

```diff
@@ -0,0 +1,168 @@
+
+Repeat the above steps one more time to add an additional DataStore with the same parameters, but this time, use `user` for the "user" parameter.
+
+!["GeoMesa DataStore configuration with \"user\" user"](/img/tutorials/2015-03-30-geomesa-feature-level-visibility/user-config.png)
+
+With your layers added in GeoServer, we're nearly ready to visualize the data. One final step is adding our custom SLD that will style your features to make visualizations of the data even easier to understand. [You can find it here.](https://raw.githubusercontent.com/geomesa/geomesa-quickstart/feature_level_visibility/featureLevelVisibility.xml)
+
+Download this XML file, or copy the contents, and add it as a Style in GeoServer. It will style the points on a map based on the visibility attribute present.
+
+Lastly, click on "Layer Preview" in the left hand sidebar and find your two newly added layers. If everything went correctly, you should see fewer results returning in the `user` layer than in the `admin` layer, and this is expected behavior. Because `user` has only been granted permission to view features with the `user` visibility, only those 500 records are returned. However, the `admin` user is granted permission to see both `admin` and `user` visibilities.
+
+## Expanding this concept
+
+In this very simple example, you wrote features of two different visibilities, added two new users to Accumulo, and granted them separate authorization levels to be able to view portions of the data.
+This tutorial has real-world use cases in security and data integrity. For example, when storing data at differing classification levels in the same database, and having users at varying levels of security,
+this ensures that sensitive data is not leaked to a user of a lower level of security.
+
+The concept of feature level visibility can be extended and modified to have many more, or only a few, visibility levels. And with GeoServer being flexible and extensible, writing a module to consider feature
+level security in GeoServer is relatively painless.
+
+At CCRi, we've implemented a very small, example Java class that does a simple match on the username provided by GeoServer. Our example class then
+matches the username to a level of authorization specified in the class and hands this off to an AuthorizationsProvider in GeoMesa. This AuthorizationsProvider uses
+the correct level of authorization when querying Accumulo, no matter the user that is specified in the GeoServer DataStore. This allows us to have only one layer per
+DataSource, rather than what is in the example above, that many users of varying authorizations levels can hit and return only the data granted to their level of security.
+
+If you're interested in learning more about this, feel free to email [jdk2pq@ccri.com](mailto:jdk2pq@ccri.com).
```

Comment:
Should probably put the geomesa-users or dev list here instead of your personal email.